### PR TITLE
workspace: image/file drop/paste

### PR DESCRIPTION
**Pull request by tvanderstad**
- moves the markdown editor in workspace (next to the svg editor)
- removes to/from host/egui clipboard mechanisms
- removes markdown editor custom events
- adds `workspace_rs::Event` custom events with support for image/file drop/paste and existing markdown editor events
- emit custom events for image paste on windows and linux; receive in markdown editor and svg editor
- change apple ffi functions so that apple clients store clipboard contents in swift between clipboard changes and pastes

fixes https://github.com/lockbook/lockbook/issues/2347

### CODE REVIEW

**Review by ad-tra — CHANGES_REQUESTED** [review]
unfortunately doesn't work on wayland

File: libs/content/workspace-ffi/src/apple/macos_ffi.rs [hunk 1]
```diff
@@ -19,16 +19,8 @@ pub unsafe extern "C" fn key_event(
...
-    if pressed && key == NSKeys::V && modifiers.command {
-        let clip = obj.from_host.clone().unwrap_or_default();
-        obj.raw_input.events.push(Event::Paste(clip));
-        clip_event = true
-    }
-
     // Event::Text
-    if !clip_event && pressed && (modifiers.shift_only() || modifiers.is_none()) && key.valid_text()
-    {
+    if pressed && (modifiers.shift_only() || modifiers.is_none()) && key.valid_text() {
```
Comment (Tremelune) [inline libs/content/workspace-ffi/src/apple/macos_ffi.rs:23]: Lotta parens and operators here...Might be worth a variable for readability:
`let allowed_modifier = modifiers.shift_only() || modifiers.is_none()
if pressed && allowed_modifier && key.valid_text() {`

**Comment by Parth** [comment]
@ad-tra do text clipboard events work?

File: libs/content/workspace-ffi/src/apple/macos_ffi.rs (hunk 1 above)
Comment (tvanderstad) [reply libs/content/workspace-ffi/src/apple/macos_ffi.rs]: Took your advice, thanks for the input!

**Review by Parth — APPROVED** [review]
@ad-tra  we're going to merge his work because it's not related to this PR, fixing this is more-so a "do we release new linux integration this release or old one"

**Pull request merged**